libs/server/Resp/Objects/SortedSetCommands.cs

```diff
@@ -827,5 +827,61 @@ private unsafe bool SortedSetDifference<TGarnetApi>(ref TGarnetApi storageApi)
 
             return true;
         }
+
+        /// <summary>
+        ///  Computes a difference operation  between the first and all successive sorted sets and store
+        ///  and returns the result to the client.
+        ///  The total number of input keys is specified.
+        /// </summary>
+        /// <param name="storageApi"></param>
+        /// <returns></returns>
+        /// <exception cref="GarnetException"></exception>
+        private unsafe bool SortedSetDifferenceStore<TGarnetApi>(ref TGarnetApi storageApi)
+             where TGarnetApi : IGarnetApi
+        {
+            if (parseState.Count < 3)
+            {
+                return AbortWithWrongNumberOfArguments("ZDIFFSTORE");
+            }
+
+            //number of keys
+            if (!parseState.TryGetInt(1, out var nKeys))
+            {
+                while (!RespWriteUtils.WriteError(CmdStrings.RESP_ERR_GENERIC_VALUE_IS_NOT_INTEGER, ref dcurr, dend))
+                    SendAndReset();
+                return true;
+            }
+
+            if (parseState.Count - 2 != nKeys)
+            {
+                while (!RespWriteUtils.WriteError(CmdStrings.RESP_SYNTAX_ERROR, ref dcurr, dend))
+                    SendAndReset();
+                return true;
+            }
+
+            var destination = parseState.GetArgSliceByRef(0);
+            var keys = new ArgSlice[nKeys];
```

Comment:
There's no reason to allocate another array for these keys as they already exist in the parse state.
You can create the ObjectInput instance here and hand that off to the storageApi instead.

Reply:
Makes sense, the only reason I did is because the existing `SortedSetDifference` method does the same thing. Instead of using `ObjectInput` I used `Span<ArgSlice>` because `Span<ArgSlice>` will be easier to use if some one need to call this programmatically using GarnetApi. Let me know if you are fine with this change.

Reply:
Yes you're correct, that's because we didn't have ObjectInput when SSD was written ;) At some point it needs to be updated as well. And yes, your solution seems fine to me.

Reply:
> And yes, your solution seems fine to me.

Marking this as resolved